reana_commons/email.py

```diff
@@ -35,13 +38,32 @@ def send_email(
     login_email=REANA_EMAIL_LOGIN,
     sender_email=REANA_EMAIL_SENDER,
 ):
-    """Send emails from REANA platform."""
+    """Send emails from REANA platform.
+
+    :param receiver_email: Email address of the receiver.
+    :param subject: Subject of the email.
+    :param body: Body of the email.
+    :param login_email: Email address of the logged user.
+    :param sender_email: Email address of the sender.
+    :raises REANAEmailNotificationError: If email cannot be sent, e.g. due to
+        missing configuration.
+    """
     message = EmailMessage()
     message["From"] = f"REANA platform <{sender_email}>"
     message["To"] = receiver_email
     message["Subject"] = subject
     message.set_content(body)
 
+    if not REANA_NOTIFICATIONS_ENABLED:
+        logging.info(
```

Comment:
We can keep this like this for now, but for the future we should improve this as we are mixing the logic to send emails (e.g. server connection, message encoding, etc.) with REANA's notification logic (e.g. do not send emails if notifications are disabled). Not only that, the behaviour of this function depends a lot on global state, thus making this function hard to test and hard to use properly.

On another note, what do you think about throwing an exception instead of just logging this? We already have `REANAEmailNotificationError`, and we are already catching it in most places, except for `_send_confirmation_email`. In this way, we also avoid having to check `REANA_NOTIFICATIONS_ENABLED` to understand if the email was actually sent or not